C compilation error when accessing an element in a multi-dimensional array of optionals

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->

<details>
<summary>V version: V 0.4.9 eecaa64, press to see full `v doctor` output</summary>

|V full version      |V 0.4.9 eecaa64
|:-------------------|:-------------------
|OS                  |linux, Linux version 6.12.11-200.fc41.x86_64 (mockbuild@8c05b49e2e66460390f7ce4d04d4f464) (gcc (GCC) 14.2.1 20250110 (Red Hat 14.2.1-7), GNU ld version 2.43.1-5.fc41) #1 SMP PREEMPT_DYNAMIC Fri Jan 24 04:59:58 UTC 2025
|Processor           |12 cpus, 64bit, little endian, AMD Ryzen 5 1600X Six-Core Processor
|Memory              |1.83GB/7.67GB
|                    |
|V executable        |/home/jacob/.local/v/v
|V last modified time|2025-02-05 08:29:54
|                    |
|V home dir          |OK, value: /home/jacob/.local/v
|VMODULES            |OK, value: /home/jacob/.vmodules
|VTMP                |OK, value: /tmp/v_1000
|Current working dir |OK, value: /home/jacob/code/v/chess
|                    |
|Git version         |git version 2.48.1
|V git status        |Error: fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
|.git/config present |false
|                    |
|cc version          |cc (GCC) 14.2.1 20250110 (Red Hat 14.2.1-7)
|gcc version         |gcc (GCC) 14.2.1 20250110 (Red Hat 14.2.1-7)
|clang version       |clang version 19.1.7 (Fedora 19.1.7-1.fc41)
|tcc version         |tcc version 0.9.28rc 2024-07-31 HEAD@1cee0908 (x86_64 Linux)
|tcc git status      |N/A
|emcc version        |N/A
|glibc version       |ldd (GNU libc) 2.40

</details>

**What did you do?**
`v bug.v`
```v
struct Test{
	board   [8][8]?Piece
}
struct Piece{
	white   bool
}
fn main() {
	t := Test{}
	println(t.board[1][2]) // error happens here
}

```

**What did you see?**
```
================== C compilation error (from tcc): ==============
cc: /tmp/v_1000/bug.01JKZECDNZ1V3W4D7F3XZ7HFNX.tmp.c:1050: error: ';' expected (got "Array_fixed_Array_fixed__option_main__Piece_8_8")
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error: 
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```

**What did you expect to see?**

no C compilation error

It looks like accessing an element in a multi-dimensional array of optionals causes this error. The error doesn't happen on 1-dimensional arrays or even multi-dimensional slices.

> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -6610,6 +6610,7 @@ fn (mut g Gen) write_types(symbols []&ast.TypeSymbol) {
 											base)};')
 									}
 								}
+								g.type_definitions.writeln('typedef ${fixed_elem_name} ${styp} [${len}];')
 							} else if !(elem_sym.info is ast.ArrayFixed && elem_sym.info.is_fn_ret) {
 								g.type_definitions.writeln('typedef ${fixed_elem_name} ${styp} [${len}];')
 							}

--- a/vlib/v/tests/options/option_nested_fixed_array_test.v
+++ b/vlib/v/tests/options/option_nested_fixed_array_test.v
@@ -1,0 +1,35 @@
+struct Test {
+	board [2][2]?Piece
+}
+
+struct Piece {
+	white bool
+}
+
+fn test_main() {
+	t := Test{
+		board: [[?Piece{
+			white: false
+		}, ?Piece{
+			white: false
+		}]!, [?Piece{
+			white: true
+		}, ?Piece{
+			white: true
+		}]!]!
+	}
+	assert '${t.board[1][1]}' == 'Option(Piece{
+    white: true
+})'
+	assert t.str() == 'Test{
+    board: [[Option(Piece{
+    white: false
+}), Option(Piece{
+    white: false
+})], [Option(Piece{
+    white: true
+}), Option(Piece{
+    white: true
+})]]
+}'
+}

